include/seqan3/range/view/kmer_hash.hpp

```diff
@@ -43,14 +43,14 @@ class kmer_hash_fn : public pipable_adaptor_base<kmer_hash_fn>
      */
     template <std::ranges::ViewableRange urng_t>
     //!\cond
-        requires Semialphabet<reference_t<urng_t>>
+        requires Semialphabet<reference_t<remove_cvref_t<urng_t>>>
     //!\endcond
     static auto impl(urng_t && urange, size_t const k)
     {
         return std::forward<urng_t>(urange) | ranges::view::sliding(k) | std::view::transform(
         [] (auto const in)
         {
-            std::hash<decltype(in)> h{};
+            std::hash<remove_cvref_t<urng_t>> h{};
```

Comment:
why this change? `in` is an alphabet but you are hashing with the specialisation of a range 🤔